### PR TITLE
feat(spec-renderer): create component [khcp-5936]

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

Create wrapper `SpecRenderer` component and move all spec display components into one package.
Addresses [KHCP-5936](https://konghq.atlassian.net/browse/KHCP-5936).

## PR Checklist

* [x] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [x] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [x] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [x] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

[KHCP-5936]: https://konghq.atlassian.net/browse/KHCP-5936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/portal/spec-renderer/docs/spec-details.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# @kong-ui-public/spec-details
+# spec-details
```
Comment (adamdehaven) [inline packages/portal/spec-renderer/docs/spec-details.md:1]:
```suggestion
# `SpecDetails.vue`
```

File: packages/portal/spec-renderer/docs/spec-details.md [hunk 2]
```diff
@@ -31,18 +31,18 @@ A Kong UI component for displaying API specs
...
 
 ```sh
-yarn add @kong-ui-public/spec-details
+yarn add @kong-ui-public/spec-renderer
 ```
 
 ### Register
 
-You can register `portal-spec-details` globally or locally.
+You can register `spec-details` globally or locally.
```
Comment (adamdehaven) [inline packages/portal/spec-renderer/docs/spec-details.md:39]: suggestion: We should remove all global registration instructions for this package and the README files; we want all of these components only imported/registered where they are being used.

File: packages/portal/spec-renderer/src/components/SpecRenderer.vue [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    requiredProps.forEach((requiredProp: string) => {
+      if (!item[requiredProp as keyof typeof item]) {
+        isValid = false
+      }
+    })
+  })
+
+  return isValid
+}
+</script>
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply packages/portal/spec-renderer/src/components/SpecRenderer.vue:96]: It's used in a prop validator

File: packages/portal/spec-renderer/src/swagger-ui-web-component.d.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 declare module '@kong-ui-public/swagger-ui-web-component' {
-  // Operation has to exist here until swagger-ui-web-component is typed
-  export interface Operation {
+  // OperationListItem has to exist here until swagger-ui-web-component is typed
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply packages/portal/spec-renderer/src/swagger-ui-web-component.d.ts:2]: I think so, I guess we hope eventually the web component will hold these type defs so we can remove them here?

File: packages/portal/spec-renderer/src/components/SpecDetails.vue [hunk 5]
```diff
@@ -67,9 +68,12 @@ const hasRequiredProps = computed((): boolean => {
 })
 
 function showAndScrollToOperation() {
+  console.log('in the function', props.activeOperation)
   if (!swaggerRef.value) {
+    console.log('no ref')
     return
   }
+  console.log('we have ref')
```
[the remark this answers is not in the document]
Comment (kaiarrowood) [reply packages/portal/spec-renderer/src/components/SpecDetails.vue]: Yes, removed

**Review by aanchalm01 — DISMISSED** [review]
Functionally tested, it's looking great! Awesome work @kaiarrowood  

image 963
image 964
image 965

**Review by adamdehaven — APPROVED** [review]
lgtm, much cleaner 🚀

**Pull request merged**